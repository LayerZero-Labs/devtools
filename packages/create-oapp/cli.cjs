#!/usr/bin/env node --max-old-space-size=4096

import("./dist/index.js")
