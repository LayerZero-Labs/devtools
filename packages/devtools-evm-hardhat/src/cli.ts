import { types as builtInTypes } from 'hardhat/config'
import { HardhatError } from 'hardhat/internal/core/errors'
import { ERRORS } from 'hardhat/internal/core/errors-list'
import type { CLIArgumentType } from 'hardhat/types'
import { splitCommaSeparated } from '@layerzerolabs/devtools'
import { isEVMAddress } from '@layerzerolabs/devtools-evm'
import { isLogLevel, LogLevel } from '@layerzerolabs/io-devtools'

/**
 * Hardhat CLI type for a comma separated list of arbitrary strings
 */
const csv: CLIArgumentType<string[]> = {
    name: 'csv',
    parse(name: string, value: string) {
        return splitCommaSeparated(value)
    },
    validate() {},
}

/**
 * Hardhat CLI type for a log level argument
 *
 * @see {@link LogLevel}
 */
const logLevel: CLIArgumentType<LogLevel> = {
    name: 'logLevel',
    parse(name: string, value: string) {
        if (!isLogLevel(value)) {
            throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
                value,
                name: name,
                type: 'logLevel',
            })
        }

        return value
    },
    validate() {},
}

/**
 * Hardhat CLI type for a function argument.
 *
 * This is only to be used with subtasks since you cannot pass functions
 * to tasks (unless you're insane and want to inline a function)
 */
export const fn: CLIArgumentType<string> = {
    name: 'function',
    parse: (argName, value) => {
        if (typeof value !== 'function') {
            throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
                value,
                name: argName,
                type: fn.name,
            })
        }

        return value
    },
    validate() {},
}

/**
 * Signer-specific CLI argument (either a non-negative index
 * or a signer EVM address)
 */
export const signer: CLIArgumentType<string | number> = {
    name: 'signer',
    parse: (argName, value) => {
        if (isEVMAddress(value)) {
            return value
        }

        const parsed = parseInt(value, 10)
        if (isNaN(parsed)) {
            throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
                value,
                name: argName,
                type: signer.name,
            })
        }

        if (parsed < 0) {
            throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
                value,
                name: argName,
                type: signer.name,
            })
        }

        return parsed
    },
    validate() {},
}

export const types = { csv, logLevel, fn, signer, ...builtInTypes }
