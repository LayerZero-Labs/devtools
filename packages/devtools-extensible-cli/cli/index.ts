export * from './AptosEVMCli'
export * from './types/NewOperation'
