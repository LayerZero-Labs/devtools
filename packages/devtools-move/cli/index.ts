export * from './init'

// EVM Operations
export { NewOperation as EVMQuoteSendOperation } from './operations/evm-quote-send'
export { NewOperation as EVMSendOperation } from './operations/evm-send'
export { NewOperation as EVMTransactionParserOperation } from './operations/evm-transaction-parser'
export { NewOperation as EVMWireOperation } from './operations/evm-wire'

// Move Operations
export { NewOperation as MoveBuildOperation } from './operations/move-build'
export { NewOperation as MoveDeployOperation } from './operations/move-deploy'
export { NewOperation as MoveSetDelegateOperation } from './operations/move-set-delegate'
export { NewOperation as MoveWireOperation } from './operations/move-wire'
export { NewOperation as MoveTransferOAppOwnerOperation } from './operations/transfer-oapp-owner'
export { NewOperation as MoveTransferObjectOwnerOperation } from './operations/transfer-object-owner'
