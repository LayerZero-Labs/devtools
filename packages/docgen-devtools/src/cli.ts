#!/usr/bin/env node

/**
 * CLI for LayerZero contract documentation generation
 */

import * as fs from 'fs'
import * as path from 'path'
import { createModuleLogger } from '@layerzerolabs/io-devtools'
import { detectContractPackages, collectContracts, generateAllDocs, generateDocumentationReadme } from './index'

async function main() {
    const args = process.argv.slice(2)
    const specificPackage = args[0]
    const logger = createModuleLogger('docgen')

    logger.info('🚀 LayerZero Contract Documentation Generator')
    logger.info('============================================')
    logger.info('')

    const rootDir = process.cwd()

    // Ensure we're in the root directory
    if (!fs.existsSync(path.join(rootDir, 'package.json')) || !fs.existsSync(path.join(rootDir, 'packages'))) {
        logger.error('❌ Error: This script must be run from the devtools root directory')
        process.exit(1)
    }

    try {
        // Detect packages
        logger.info('🔍 Detecting EVM contract packages...')
        let packages = detectContractPackages(rootDir)

        if (packages.length === 0) {
            logger.error('❌ No contract packages detected')
            process.exit(1)
        }

        // Filter to specific package if requested
        if (specificPackage) {
            const found = packages.find((p) => p.name === specificPackage)
            if (!found) {
                logger.error(`❌ Package '${specificPackage}' not found in detected contract packages`)
                logger.error('Available packages:')
                packages.forEach((p) => logger.error(`  - ${p.name}`))
                process.exit(1)
            }
            packages = [found]
            logger.info(`📦 Processing only: ${specificPackage}`)
        } else {
            logger.info(`📦 Processing all ${packages.length} detected packages`)
        }

        // Set up paths
        const outputDir = path.join(rootDir, 'docgen-out')
        const templatesDir = path.resolve(__dirname, '..', 'templates')

        // Check if templates exist
        if (!fs.existsSync(templatesDir)) {
            logger.error(`❌ Templates directory not found at ${templatesDir}`)
            logger.error('Please ensure documentation templates are set up first.')
            process.exit(1)
        }

        // Generate documentation
        await generateAllDocs(packages, {
            rootDir,
            outputDir,
            templatesDir,
        })

        // Generate inventory and summary
        logger.info('')
        logger.info('📊 Generating contract inventory...')
        const { inventory, packages: allPackages } = collectContracts(rootDir)

        // Write inventory
        fs.writeFileSync(path.join(outputDir, 'contract-inventory.json'), JSON.stringify(inventory, null, 2))

        // Write README
        const readmeContent = generateDocumentationReadme(inventory, allPackages)
        fs.writeFileSync(path.join(outputDir, 'README.md'), readmeContent)

        logger.info('')
        logger.info('✅ Documentation generation complete!')
        logger.info('')
        logger.info(`📁 Output location: ${outputDir}`)
        logger.info('')
        logger.info('📦 Packages documented:')
        packages.forEach((p) => logger.info(`  - ${p.name}`))
        logger.info('')
        logger.info('Next steps:')
        logger.info(`1. Review the generated documentation in ${outputDir}`)
        logger.info('2. Copy docgen-out/ to your Docusaurus docs repository')
        logger.info('3. Add the contract reference section to your Docusaurus sidebar')
        logger.info('')
    } catch (error) {
        logger.error('❌ Error:', error)
        process.exit(1)
    }
}

// Run the CLI
main().catch((error) => {
    const logger = createModuleLogger('docgen')
    logger.error('Fatal error:', error)
    process.exit(1)
})
