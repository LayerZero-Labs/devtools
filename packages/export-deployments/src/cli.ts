import { Command } from 'commander'
import {
    deploymentFilesOption,
    deploymentsPathOption,
    generatorOption,
    logLevelOption,
    networksOption,
    outputPathOption,
    skipDeploymentFilesOption,
} from '@/cli/options'
import { COLORS, SUCCESS_SYMBOL } from '@/common/logger'
import { createIncludeDirent, generateSafe } from '@/index'
import { pipe } from 'fp-ts/lib/function'
import * as E from 'fp-ts/Either'
import type { CodeGenerator } from '@/generator/types'
import { type LogLevel, createLogger, pluralizeNoun } from '@layerzerolabs/io-devtools'
import { version } from '../package.json'

// Ensure we exit with zero code on SIGTERM/SIGINT
//
// This is important for prompts
const handleSigTerm = () => process.exit(0)
process.on('SIGINT', handleSigTerm)
process.on('SIGTERM', handleSigTerm)

new Command('export-deployments')
    .version(version)
    .addOption(deploymentsPathOption)
    .addOption(outputPathOption)
    .addOption(logLevelOption)
    .addOption(networksOption)
    .addOption(deploymentFilesOption)
    .addOption(skipDeploymentFilesOption)
    .addOption(generatorOption)
    .action(
        async (args: {
            deployments: string
            files?: string[]
            exclude?: string[]
            networks?: string[]
            logLevel: LogLevel
            outDir: string
            generator: CodeGenerator
        }) => {
            const defaultLogger = createLogger(args.logLevel)

            defaultLogger.debug(COLORS.default`Exporting deployments from ${args.deployments} to ${args.outDir}`)

            pipe(
                generateSafe({
                    deploymentsDir: args.deployments,
                    outDir: args.outDir,
                    includeDeploymentFile: createIncludeDirent(args.files, args.exclude),
                    includeNetworkDir: createIncludeDirent(args.networks),
                    generator: args.generator,
                }),
                E.foldW(
                    (error) => {
                        defaultLogger.error(COLORS.error`Got an error during code generation: ${error}`)

                        process.exit(1)
                    },
                    (outputFiles) => {
                        const numGeneratedFiles = outputFiles.length

                        if (numGeneratedFiles === 0) {
                            defaultLogger.info(COLORS.default`No files generated, exiting`)
                        } else {
                            const message = pluralizeNoun(
                                numGeneratedFiles,
                                `Generated 1 file:`,
                                `Generated ${numGeneratedFiles} files:`
                            )

                            defaultLogger.info(COLORS.default`${SUCCESS_SYMBOL} ${message}`)
                            for (const { path } of outputFiles) {
                                defaultLogger.info(COLORS.default`\t${path}`)
                            }
                        }

                        process.exit(0)
                    }
                )
            )
        }
    )
    .parseAsync()
