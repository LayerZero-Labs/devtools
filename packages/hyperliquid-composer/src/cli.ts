import { Command } from 'commander'
import { LogLevel } from '@layerzerolabs/io-devtools'
import {
    setBlock,
    requestEvmContract,
    finalizeEvmContract,
    coreSpotDeployment,
    intoAssetBridgeAddress,
    hipTokenInfo,
    tradingFee,
    userGenesis,
    genesis,
    createSpotDeployment,
    registerTradingSpot,
    spotDeployState,
} from './commands'

const program = new Command()

program.name('oft-hyperliquid-evm').description('CLI tools for HyperLiquid OFT operations')

program
    .command('to-bridge')
    .description('Returns the asset bridge address for a token index')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .action(intoAssetBridgeAddress)

program
    .command('set-block')
    .description('Set block size')
    .requiredOption('-s, --size <size>', 'Block size (big/small)')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'Private Key')
    .action(setBlock)

program
    .command('request-evm-contract')
    .description('Set the core spot to connect to an EVM contract')
    .option('-oapp, --oapp-config <oapp-config>', 'OAPP config')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'HyperCore Asset deployer private key')
    .action(requestEvmContract)

program
    .command('finalize-evm-contract')
    .description('Finalize the EVM contract')
    .option('-oapp, --oapp-config <oapp-config>', 'OAPP config')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'Private key')
    .action(finalizeEvmContract)

program
    .command('core-spot')
    .description('Get core spot metadata information')
    .option('-a, --action <action>', 'Action (create/get)', 'get')
    .option('-oapp, --oapp-config <oapp-config>', 'OAPP config')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .action(coreSpotDeployment)

program
    .command('hip-token')
    .description('Get information about a HyperCore HIP-1 token')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .action(hipTokenInfo)

program
    .command('spot-deploy-state')
    .description('Get the current deployment state of a hypercore token')
    .requiredOption(
        '-idx, --token-index <token-index>',
        'Filter on token index. To view all deployments, use "--log-level verbose"'
    )
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-da, --deployer-address <0x>', 'Core spot deployer address (optional)')
    .action(spotDeployState)

program
    .command('trading-fee')
    .description('Set trading fee share')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-s, --share <share>', 'Share')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'Private key')
    .action(tradingFee)

program
    .command('user-genesis')
    .description('Set user genesis')
    .option('-a, --action <action>', 'Action (userAndWei/existingTokenAndWei/blacklistUsers)', '*')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'Private key')
    .action(async (options) => {
        const validActions = ['*', 'userAndWei', 'existingTokenAndWei', 'blacklistUsers']
        if (!validActions.includes(options.action)) {
            throw new Error(`Invalid action: ${options.action}. Valid actions are: ${validActions.join(', ')}`)
        }
        await userGenesis(options)
    })

program
    .command('set-genesis')
    .description('Set genesis')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'Private key')
    .action(genesis)

program
    .command('create-spot-deployment')
    .description('Create a spot deployment without hyperliquidity')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'Private key')
    .action(createSpotDeployment)

program
    .command('register-spot')
    .description('Register trading spot against USDC')
    .requiredOption('-idx, --token-index <token-index>', 'Token index')
    .requiredOption('-n, --network <network>', 'Network (mainnet/testnet)')
    .option('-l, --log-level <level>', 'Log level', LogLevel.info)
    .option('-pk, --private-key <0x>', 'Private key')
    .action(registerTradingSpot)

program.parse()
