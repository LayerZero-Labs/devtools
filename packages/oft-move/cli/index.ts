export * from './init'
// OFT Move Operations
export * from './operations'
