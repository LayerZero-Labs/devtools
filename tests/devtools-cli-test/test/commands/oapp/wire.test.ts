import 'hardhat'
import { isFile } from '@layerzerolabs/io-devtools'
import { dirname, join, relative } from 'path'
import { cwd } from 'process'
import { deployContract, setupDefaultEndpointV2 } from '@layerzerolabs/test-setup-devtools-evm-hardhat'
import { runCli } from '../../__utils__/cli'

describe(`command oapp wire`, () => {
    const runWire = (args: string[] = []) => runCli(['oapp', 'wire', ...args])

    const CONFIGS_BASE_DIR = relative(cwd(), join(__dirname, '__data__', 'configs'))
    const SETUPS_BASE_DIR = relative(cwd(), join(__dirname, '__data__', 'setups'))

    const configPathFixture = (fileName: string): string => {
        const path = join(CONFIGS_BASE_DIR, fileName)

        expect(isFile(path)).toBeTruthy()

        return path
    }

    const setupPathFixture = (fileName: string): string => {
        const path = join(SETUPS_BASE_DIR, fileName)

        expect(isFile(path)).toBeTruthy()

        return path
    }

    beforeAll(async () => {
        await deployContract('EndpointV2')
        await setupDefaultEndpointV2()
    })

    describe('with missing options', () => {
        it('should fail if the setup file is not passed', async () => {
            const result = runWire(['--oapp-config', './does-not-exist.js'])

            expect(result.stderr).toMatch("required option '-s,--setup <path>' not specified")
            expect(result.status).toBe(1)
        })

        it('should fail if the config file is not passed', async () => {
            const result = runWire(['--setup', './does-not-exist.js'])

            expect(result.stderr).toMatch("required option '--oapp-config <path>' not specified")
            expect(result.status).toBe(1)
        })
    })

    describe('with invalid options', () => {
        beforeAll(async () => {
            await deployContract('OApp')
        })

        describe('with invalid config option', () => {
            it('should fail if the config file does not exist', async () => {
                const result = runWire([
                    '--oapp-config',
                    './does-not-exist.js',
                    '--setup',
                    setupPathFixture('valid.setup.ts'),
                ])

                expect(result.stdout).toMatch(`Unable to read config file './does-not-exist.js'`)
                expect(result.status).not.toBe(0)
            })

            it('should fail if the config file is not a file', async () => {
                const oappConfig = dirname(configPathFixture('invalid.config.empty.json'))

                const result = runWire(['--oapp-config', oappConfig, '--setup', setupPathFixture('valid.setup.ts')])
                expect(result.stdout).toMatch(`Unable to read config file '${oappConfig}`)
                expect(result.status).not.toBe(0)
            })

            it('should fail if the config file is not a valid JSON or JS file', async () => {
                const oappConfig = 'README.md'

                expect(isFile(oappConfig)).toBeTruthy()

                const result = runWire(['--oapp-config', oappConfig, '--setup', setupPathFixture('valid.setup.ts')])
                expect(result.stdout).toMatch(`Unable to read config file '${oappConfig}`)
                expect(result.status).not.toBe(0)
            })

            it('should fail with an empty JSON file', async () => {
                const oappConfig = configPathFixture('invalid.config.empty.json')

                const result = runWire(['--oapp-config', oappConfig, '--setup', setupPathFixture('valid.setup.ts')])
                expect(result.stdout).toMatch(`Unable to read config file '${oappConfig}'`)
                expect(result.stdout).toMatch(`Unexpected end of JSON input`)
                expect(result.status).not.toBe(0)
            })

            it('should fail with an empty JS file', async () => {
                const oappConfig = configPathFixture('invalid.config.empty.js')

                const result = runWire(['--oapp-config', oappConfig, '--setup', setupPathFixture('valid.setup.ts')])
                expect(result.stdout).toMatchSnapshot()
                expect(result.status).not.toBe(0)
            })

            it('should fail with a malformed JS file (001)', async () => {
                const oappConfig = configPathFixture('invalid.config.001.js')

                const result = runWire(['--oapp-config', oappConfig, '--setup', setupPathFixture('valid.setup.ts')])
                expect(result.stdout).toMatchSnapshot()
                expect(result.status).not.toBe(0)
            })

            it('should fail with a misconfigured file (001)', async () => {
                const oappConfig = configPathFixture('valid.config.misconfigured.001.js')

                const result = runWire(['--oapp-config', oappConfig, '--setup', setupPathFixture('valid.setup.ts')])
                expect(result.stdout).toMatchSnapshot()
                expect(result.status).not.toBe(0)
            })
        })

        describe('with invalid setup option', () => {
            it('should fail if the setup file does not exist', async () => {
                const result = runWire([
                    '--oapp-config',
                    configPathFixture('valid.config.empty.js'),
                    '--setup',
                    './does-not-exist.js',
                ])

                expect(result.stdout).toMatch(`Unable to read setup file './does-not-exist.js'`)
                expect(result.status).not.toBe(0)
            })

            it('should fail if the config file is not a file', async () => {
                const setup = dirname(configPathFixture('invalid.config.empty.json'))

                const result = runWire(['--oapp-config', configPathFixture('valid.config.empty.js'), '--setup', setup])
                expect(result.stdout).toMatch(`Unable to read setup file '${setup}`)
                expect(result.status).not.toBe(0)
            })

            it('should fail if the config file is not a valid JSON or JS file', async () => {
                const setup = 'README.md'

                expect(isFile(setup)).toBeTruthy()

                const result = runWire(['--oapp-config', configPathFixture('valid.config.empty.js'), '--setup', setup])
                expect(result.stdout).toMatch(`Unable to read setup file '${setup}`)
                expect(result.status).not.toBe(0)
            })

            it('should fail with an empty JS file', async () => {
                const setup = setupPathFixture('invalid.setup.empty.js')

                const result = runWire(['--oapp-config', configPathFixture('valid.config.empty.js'), '--setup', setup])
                expect(result.stdout).toMatchSnapshot()
                expect(result.status).not.toBe(0)
            })

            it('should fail with a malformed JS file (001)', async () => {
                const setup = setupPathFixture('invalid.setup.001.js')

                const result = runWire(['--oapp-config', configPathFixture('valid.config.empty.js'), '--setup', setup])
                expect(result.stdout).toMatchSnapshot()
                expect(result.status).not.toBe(0)
            })
        })
    })

    describe('with valid configs', () => {
        beforeEach(async () => {
            await deployContract('OApp')
        })

        it('should exit with an empty config', async () => {
            const oappConfig = configPathFixture('valid.config.empty.ts')
            const setup = setupPathFixture('valid.setup.ts')

            const result = runWire(['--oapp-config', oappConfig, '--setup', setup])
            expect(result.stdout).toMatch(`The OApp is wired, no action is necessary`)
            expect(result.status).toBe(0)
        })

        it('should not understand hardhat config', async () => {
            const oappConfig = configPathFixture('valid.config.connected.js')
            const setup = setupPathFixture('valid.setup.ts')

            const result = runWire(['--oapp-config', oappConfig, '--setup', setup])
            expect(result.stderr).toMatchSnapshot()
            expect(result.status).not.toBe(0)
        })
    })
})
