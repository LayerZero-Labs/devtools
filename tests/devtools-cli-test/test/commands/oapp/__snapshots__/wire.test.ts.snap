// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command oapp wire with invalid options with invalid config option should fail with a malformed JS file (001) 1`] = `
"





                                   **********
                                 **************
                               ******************
                              ********************
                              *********  *********
                              *********  *********
                              *********  *********
                                 ******  *********
                               ********  *********
                              *********  ********
                              *********  ******
                              *********  *********
                              *********  *********
                              *********  *********
                              ********************
                               ******************
                                ****************
                                   **********







[33mwarn[39m:    This command is just a placeholder. Please use @layerzerolabs/toolbox-hardhat package for the time being.
[32minfo[39m:    Loading config file from test/commands/oapp/__data__/configs/invalid.config.001.js
[31merror[39m:   Failed to load config from test/commands/oapp/__data__/setups/valid.setup.ts:

Error: Config from file 'test/commands/oapp/__data__/configs/invalid.config.001.js' is malformed. Please fix the following errors:

Property 'contracts.0.point': Required
Property 'contracts.0.config': Required
Property 'contracts.1.point': Required
Property 'contracts.1.config': Required
Property 'connections': Required
"
`;

exports[`command oapp wire with invalid options with invalid config option should fail with a misconfigured file (001) 1`] = `
"





                                   **********
                                 **************
                               ******************
                              ********************
                              *********  *********
                              *********  *********
                              *********  *********
                                 ******  *********
                               ********  *********
                              *********  ********
                              *********  ******
                              *********  *********
                              *********  *********
                              *********  *********
                              ********************
                               ******************
                                ****************
                                   **********







[33mwarn[39m:    This command is just a placeholder. Please use @layerzerolabs/toolbox-hardhat package for the time being.
[32minfo[39m:    Loading config file from test/commands/oapp/__data__/configs/valid.config.misconfigured.001.js
[31merror[39m:   Failed to load config from test/commands/oapp/__data__/setups/valid.setup.ts:

Error: Config from file 'test/commands/oapp/__data__/configs/valid.config.misconfigured.001.js' is malformed. Please fix the following errors:

Property 'contracts.0.point': Required
Property 'contracts.0.config': Required
Property 'contracts.1.point': Required
Property 'contracts.1.config': Required
Property 'connections.0.vector': Required
Property 'connections.0.config': Required
Property 'connections.1.vector': Required
Property 'connections.1.config': Required
"
`;

exports[`command oapp wire with invalid options with invalid config option should fail with an empty JS file 1`] = `
"





                                   **********
                                 **************
                               ******************
                              ********************
                              *********  *********
                              *********  *********
                              *********  *********
                                 ******  *********
                               ********  *********
                              *********  ********
                              *********  ******
                              *********  *********
                              *********  *********
                              *********  *********
                              ********************
                               ******************
                                ****************
                                   **********







[33mwarn[39m:    This command is just a placeholder. Please use @layerzerolabs/toolbox-hardhat package for the time being.
[32minfo[39m:    Loading config file from test/commands/oapp/__data__/configs/invalid.config.empty.js
[31merror[39m:   Failed to load config from test/commands/oapp/__data__/setups/valid.setup.ts:

Error: Config from file 'test/commands/oapp/__data__/configs/invalid.config.empty.js' is malformed. Please fix the following errors:

Property 'contracts': Required
Property 'connections': Required
"
`;

exports[`command oapp wire with invalid options with invalid setup option should fail with a malformed JS file (001) 1`] = `
"





                                   **********
                                 **************
                               ******************
                              ********************
                              *********  *********
                              *********  *********
                              *********  *********
                                 ******  *********
                               ********  *********
                              *********  ********
                              *********  ******
                              *********  *********
                              *********  *********
                              *********  *********
                              ********************
                               ******************
                                ****************
                                   **********







[33mwarn[39m:    This command is just a placeholder. Please use @layerzerolabs/toolbox-hardhat package for the time being.
[31merror[39m:   Failed to load setup from test/commands/oapp/__data__/setups/invalid.setup.001.js:

Error: Setup from file 'test/commands/oapp/__data__/setups/invalid.setup.001.js' is malformed. Please fix the following errors:

Property 'createSdk': Expected function, received array
Property 'createSigner': Required
"
`;

exports[`command oapp wire with invalid options with invalid setup option should fail with an empty JS file 1`] = `
"





                                   **********
                                 **************
                               ******************
                              ********************
                              *********  *********
                              *********  *********
                              *********  *********
                                 ******  *********
                               ********  *********
                              *********  ********
                              *********  ******
                              *********  *********
                              *********  *********
                              *********  *********
                              ********************
                               ******************
                                ****************
                                   **********







[33mwarn[39m:    This command is just a placeholder. Please use @layerzerolabs/toolbox-hardhat package for the time being.
[31merror[39m:   Failed to load setup from test/commands/oapp/__data__/setups/invalid.setup.empty.js:

Error: Setup from file 'test/commands/oapp/__data__/setups/invalid.setup.empty.js' is malformed. Please fix the following errors:

Property 'createSdk': Required
Property 'createSigner': Required
"
`;

exports[`command oapp wire with valid configs should not understand hardhat config 1`] = `""`;
